alcs-api/src/application-amendment/application-amendment.service.spec.ts

```diff
@@ -0,0 +1,274 @@
+import { AmendmentProfile } from '../common/automapper/amendment.automapper.profile';
+import { ServiceNotFoundException } from '../common/exceptions/base.exception';
+import {
+  initApplicationAmendementMockEntity,
+  initApplicationMockEntity,
+} from '../common/utils/test-helpers/mockEntities';
+import {
+  ApplicationAmendmentCreateDto,
+  ApplicationAmendmentUpdateDto,
+} from './application-amendment.dto';
+import { ApplicationAmendment } from './application-amendment.entity';
+import { ApplicationAmendmentService } from './application-amendment.service';
+
+describe('AmendmentService', () => {
+  let amendmentRepoMock: DeepMocked<Repository<ApplicationAmendment>>;
+  let service: ApplicationAmendmentService;
+  let applicationServiceMock: DeepMocked<ApplicationService>;
+  let cardServiceMock: DeepMocked<CardService>;
+
+  let mockAmendment;
+
+  const DEFAULT_RELATIONS: FindOptionsRelations<ApplicationAmendment> = {
+    application: {
+      type: true,
+      region: true,
+      localGovernment: true,
+      decisionMeetings: true,
+    },
+    card: {
+      board: true,
+      type: true,
+      status: true,
+      assignee: true,
+    },
+  };
+
+  beforeEach(async () => {
+    applicationServiceMock = createMock<ApplicationService>();
+    cardServiceMock = createMock<CardService>();
+    amendmentRepoMock = createMock<Repository<ApplicationAmendment>>();
+
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        AutomapperModule.forRoot({
+          strategyInitializer: classes(),
+        }),
+      ],
+      providers: [
+        ApplicationAmendmentService,
+        {
+          provide: ApplicationService,
+          useValue: applicationServiceMock,
+        },
+        {
+          provide: CardService,
+          useValue: cardServiceMock,
+        },
+        {
+          provide: getRepositoryToken(ApplicationAmendment),
+          useValue: amendmentRepoMock,
+        },
+        AmendmentProfile,
+      ],
+    }).compile();
+    service = module.get<ApplicationAmendmentService>(
+      ApplicationAmendmentService,
+    );
+
+    mockAmendment = initApplicationAmendementMockEntity();
+    amendmentRepoMock.findOneOrFail.mockResolvedValue(mockAmendment);
+    amendmentRepoMock.findOneByOrFail.mockResolvedValue(mockAmendment);
+    amendmentRepoMock.find.mockResolvedValue([mockAmendment]);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  it('should have correct filter condition in getByCode', async () => {
+    const fakeBoardCode = 'fake';
+    const findOptions = {
+      where: { card: { board: { code: fakeBoardCode } } },
+      relations: DEFAULT_RELATIONS,
+    };
+
+    await service.getByBoardCode(fakeBoardCode);
+
+    expect(amendmentRepoMock.find).toBeCalledWith(findOptions);
+  });
+
+  it('should successfully create application and amendment card if app does not exist', async () => {
+    const mockamendmentCreateDto = {
```

Comment:
You can move this into separate function and reuse in other places

Reply:
Good catch, done! Moved to beforeEach and re-used.